MaraBot/Core/CommandUtils.cs

```diff
@@ -84,5 +86,70 @@ public static async Task<string[]> MentionRoleWithoutPing(CommandContext ctx, Di
             var hasPerms = await HasBotPermissions(ctx, Permissions.MentionEveryone);
             return roles.Select(r => r.IsMentionable || hasPerms ? $"@({r.Name})" : r.Mention).ToArray();
         }
+
+        /// <summary>
+        /// Grants a role to a user.
+        /// </summary>
+        public static async Task GrantRoleAsync(CommandContext ctx, string roleString)
+        {
+            // Grant user their new role.
+            var newRole = ctx.Guild.Roles
+                .FirstOrDefault(kvp => roleString.Equals(kvp.Value.Name));
+
+            if (newRole.Value == null)
+            {
+                await ctx.RespondAsync(
+                    "No role set for access to spoiler channel.\n" +
+                    "This shouldn't happen! Please contact your friendly neighbourhood developers!");
+
+                throw new InvalidOperationException($"role {newRole} has not been found in guild {ctx.Guild.Name}.");
+            }
+
+            await ctx.Member.GrantRoleAsync(newRole.Value);
+        }
+
+        /// <summary>
+        /// Sends a message to a specific channel.
+        /// </summary>
+        public static async Task SendToChannelAsync(CommandContext ctx, string channelName, DiscordEmbedBuilder embed)
+        {
+            var channel = ctx.Guild.Channels
+                .FirstOrDefault(kvp => channelName.Equals(kvp.Value.Name)).Value;
+
+            if (channel == null)
+            {
+                var errorMessage = $"Channel {channelName} has not been found in guild {ctx.Guild.Name}.";
+
+                await ctx.RespondAsync(
+                    errorMessage +
+                    "This shouldn't happen! Please contact your friendly neighbourhood developers!");
+
+                throw new InvalidOperationException(errorMessage);
+            }
+
+            await channel.SendMessageAsync(embed);
+        }
+
+        /// <summary>
+        /// Verifies if current channel matches specified channel.
+        /// </summary>
+        public static async Task<bool> VerifyChannel(CommandContext ctx, string channelName)
+        {
+            var channel = ctx.Guild.Channels
+                .FirstOrDefault(kvp => channelName.Equals(kvp.Value.Name)).Value;
+
+            if (channel == null)
+            {
+                var errorMessage = $"Channel {channelName} has not been found in guild {ctx.Guild.Name}.";
+
+                await ctx.RespondAsync(
+                    errorMessage +
+                    "This shouldn't happen! Please contact your friendly neighbourhood developers!");
+
+                throw new InvalidOperationException(errorMessage);
+            }
+
+            return ctx.Channel.Equals(channel);
+        }
```

Comment:
Maybe it's better to make the helper functions just return the `channel` or `role` they are looking for.
Then the code calling the function can do the operation like:
```c#
ctx.Channel.Equals(channel);
```
This also removes even more code duplication, because the functions `SendToChannelAsync` and `VerifyChannel` only differ in the last line.

Reply:
Hmm .. not sure how much we'll be able to reuse these functions afterwards. I think we should keep them as they are for now and we can rework them if it becomes necessary.

Also, I have another branch where I added a revoke roles function. I needed a function that would revoke multiple roles at once (both "completed" and "forfeit" roles potentially) and it seems cleaner to do it all at once in a single function.